json_serializable/lib/src/utils.dart

```diff
@@ -11,6 +11,42 @@ import 'package:analyzer/src/dart/resolver/inheritance_manager.dart'
 
 import 'package:source_gen/source_gen.dart';
 
+// TODO: still need handle tripe singe/double quotes within `value`
+String escapeDartString(String value) {
+  if (value.contains('\n')) {
+    return "r'''\n$value'''";
+  }
+
+  var containsSingleQuote = value.contains("'");
+  var contains$ = value.contains(r'$');
+
+  if (containsSingleQuote) {
+    if (value.contains('"')) {
+      // `value` contains both single and double quotes as well as `$`.
+      // The only safe way to wrap the content is to escape all of the
+      // problematic characters.
+      var string = value
+          .replaceAll('\$', '\\\$')
+          .replaceAll('"', '\\"')
+          .replaceAll("'", "\\'");
```

Comment:
Doing three replacements operation, where at least two are known to change the string, may be a little costly on large strings.
On the other hand, it's probably much cheaper than the alternative:
```dart
  var string = value.replaceAll(new RegExp(r"""(?=[$'"])""", r"\");
```
(even if the `RegExp` and replacement function are both made constant).

Consider whether it might be worth it as a special case for large strings.

Also still a problem if the string contains `\$` already.

Reply:
Will investigate in a follow-up CL. This is strictly better than what we had before...I think...